### PR TITLE
samples: matter: add support for the TimeSync cluster in lock app

### PR DESCRIPTION
**Pull request by markaj-nordic**
* updated .zap file
* regenerated data model
* enabled ReadClient to support TimeSyncClient feature

- [x] Test with CHIP Tool

### CODE REVIEW

File: samples/matter/lock/prj.conf [hunk 1]
```diff
@@ -42,3 +42,6 @@ CONFIG_CHIP_APP_LOG_LEVEL=3
 # Enable Factory Data feature
 CONFIG_CHIP_FACTORY_DATA=y
 CONFIG_CHIP_FACTORY_DATA_BUILD=y
+
+# Enable read client for needed for TimeSyncClient feature
+CONFIG_CHIP_ENABLE_READ_CLIENT=y
```
Comment (ArekBalysNordic) [inline samples/matter/lock/prj.conf:47]: Is it for debugging purposes? I guess we shouldn't enable by default optional features.

**Review by ArekBalysNordic — COMMENTED** [review]
If the cluster is "heavy" we can consider moving responsibility to enable it to the user and provide proper documentation.

File: samples/matter/lock/prj.conf (hunk 1 above)
Comment (kkasperczyk-no) [reply samples/matter/lock/prj.conf:47]: The TimeSync cluster is configured as a client, so I guess it cannot work without a read client functionality.

Comment (kkasperczyk-no) [reply samples/matter/lock/prj.conf:47]: Well, after a second thought, do we really want to be a client and server? Wouldn't server role be enough for us?

Comment (markaj-nordic) [reply samples/matter/lock/prj.conf:47]: To support all required Time Sync features I believe we need to be both server and client. The latter is needed for Time Sync Client feature (the device proactively queries trusted nodes about the current time after joining the fabric).

**Comment by markaj-nordic** [comment]
This feature will be a part of https://github.com/nrfconnect/sdk-nrf/pull/15485. Closing this one.

**Pull request closed without merging**